includes/components/class-attribute.php

```diff
@@ -85,9 +85,138 @@ protected function boot() {
 
 			// Disable Jetpack search.
 			add_filter( 'jetpack_search_should_handle_query', [ $this, 'disable_jetpack_search' ], 10, 2 );
+
+			// Set search order.
+			add_filter( 'posts_orderby', [ $this, 'set_search_order' ] );
+		} else {
+
+			// Alter settings.
+			add_filter( 'hivepress/v1/settings', [ $this, 'alter_settings' ] );
 		}
 	}
 
+	/**
+	 * Sets WP search order.
+	 *
+	 * @param string $orderby Order filters.
+	 * @return string
+	 */
+	public function set_search_order( $orderby ) {
+		global $wpdb;
+
+		// Check sort option.
+		if ( isset( $_GET['_sort'] ) ) {
+			return $orderby;
+		}
+
+		foreach ( [ 'listing', 'vendor', 'request' ] as $model ) {
+
+			// Get page ID.
+			$page_id = absint( get_option( 'hp_page_' . $model . 's' ) );
+
+			// Check page.
+			if ( ! ( ( $page_id && is_page( $page_id ) ) || is_post_type_archive( 'hp_' . $model ) || ( is_tax() && strpos( get_queried_object()->taxonomy, 'hp_' . $model . '_' ) === 0 ) ) ) {
+				continue;
+			}
+
+			// Set sort param.
+			$sort_param = get_option( hp\prefix( $model . '_sorting_option' ) );
+
+			// Set sort options.
+			$order_by = '';
+			$order = '';
+
+			if ( 'title' === $sort_param ) {
+
+				// Set sort options.
+				$order_by = $wpdb->posts . '.post_title';
+				$order = 'ASC';
+			} else {
+
+				// Get sort order.
+				$sort_order = 'ASC';
+
+				if ( strpos( $sort_param, '__' ) ) {
+					list($sort_param, $sort_order) = explode( '__', $sort_param );
+				}
+
+				// Get category ID.
+				$category_id = $this->get_category_id( $model );
+
+				// Get attributes.
+				$attributes = $this->get_attributes( $model, (array) $category_id );
+
+				// Get sort attribute.
+				$sort_attribute = hp\get_array_value( $attributes, $sort_param );
+
+				if ( $sort_attribute && $sort_attribute['sortable'] ) {
+
+					// Get sort field.
+					$sort_field = hp\create_class_instance( '\HivePress\Fields\\' . $sort_attribute['edit_field']['type'], [ $sort_attribute['edit_field'] ] );
+
+					if ( $sort_field && $sort_field::get_meta( 'sortable' ) ) {
+
+						// Set sort options.
+						$order_by = $sort_param . '__order';
+						$order = strtoupper( $sort_order );
+					}
+				}
+			}
+
+			// Check sort options.
+			if ( $order_by && $order ) {
+
+				// Set sort order.
+				$orderby = $order_by . ' ' . $order;
+			}
+
+			break;
+		}
+
+		return $orderby;
+	}
+
+	/**
+	 * Alters settings.
+	 *
+	 * @param array $settings Settings configuration.
+	 * @return array
+	 */
+	public function alter_settings( $settings ) {
+		foreach ( [ 'listing', 'vendor', 'request' ] as $model ) {
```

Comment:
Please use the same solution we used in one of the recent tasks to detect if the mode is searchable, if I remember correctly we checked "has_archive", hardcoding is not ok here

Reply:
Maybe we should check has_archive in some separate callback and set some parameter to the $models class parameter (since every model can be set as a key with an array of parameters), e.g. "searchable"=>true

This way we'll avoid duplication of code where we fetch the post types config every time and check has_archive

Reply:
I couldn't found any `has_archive` checks in the code, could you provide a branch link maybe?

Reply:
It's here https://github.com/hivepress/hivepress/blob/master/includes/components/class-attribute.php#L1554 Maybe this is an old branch which wasn't in sync with the master, please press CTRL+Shift+U if you use GitHub Desktop to merge master into the task branch from time to time because there may be parallel changes.

Instead of duplicating this nested check, we may use a single callback or code in the "boot" to go through the post types and set specific "searchable" flags for models in the models array via $this->models, based on this check.